presto-main/src/main/java/io/prestosql/SystemSessionProperties.java

```diff
@@ -103,7 +103,7 @@
     public static final String MAX_RECURSION_DEPTH = "max_recursion_depth";
     public static final String USE_MARK_DISTINCT = "use_mark_distinct";
     public static final String PREFER_PARTIAL_AGGREGATION = "prefer_partial_aggregation";
-    public static final String OPTIMIZE_TOP_N_ROW_NUMBER = "optimize_top_n_row_number";
+    public static final String OPTIMIZE_TOP_N_RANKING = "optimize_top_n_ranking";
```

Comment:
We should have equivalent of `@LegacyConfig` for session toggles.

Reply:
@findepi, are you saying that we have an equivalent of @LegacyConfig, and that we should use it for this PR? Or that someone should implement an equivalent at some point?

Reply:
i am not aware that we have

Reply:
Yea, that's what I understood too. It sounds like a reasonable feature request. Would you think that this blocks any of this PR? If not, we can file an issue for that.

Reply:
I don't think it blocks here. But would be nice to have. Would you be able to implement this?

Reply:
I might take a look at this if I get some time, but no guarantees